addons/html_builder/static/src/core/builder_options_plugin.js

```diff
@@ -99,6 +102,21 @@ export class BuilderOptionsPlugin extends Plugin {
         return null;
     }
 
+    /**
+     * Deactivates the containers if the current target is not visible anymore.
+     *
+     * @param {Map<HTMLElement, boolean>} elToVisibilityMap - A map that
+     * provides information on the visibility of elements on the page.
+     */
+    deactivateContainersIfTargetHidden(elToVisibilityMap) {
+        for (const [el, visibility] of elToVisibilityMap.entries()) {
+            if (el.contains(this.target) && !visibility) {
+                this.deactivateContainers();
+                return;
+            }
+        }
+    }
```

Comment:
We need something more generic. Why not use onSelectedElVisibilityUpdated ? 
Maybe rename onElVisibilityUpdated. 
Check if el.contains(this.target) || el === this.target  && !show then  this.deactivateContainers